### PR TITLE
feat(debug): persist DevTools preferences in Chromium

### PR DESCRIPTION
**Pull request by dgozman**
When `PLAYWRIGHT_DEBUG_UI` is set, we store `devtools-preferences.json` file in the downloaded browser directory.

### CODE REVIEW

File: src/server/chromium.ts [hunk 1]
```diff
@@ -191,6 +192,11 @@ export class Chromium extends AbstractBrowserType<CRBrowser> {
       chromeArguments.push('--no-startup-window');
     return chromeArguments;
   }
+
+  private _devtools(): CRDevTools | undefined {
+    if (getFromENV('PLAYWRIGHT_DEBUG_UI'))
```
Comment (aslushnikov) [inline src/server/chromium.ts:197]: Maybe `PLAYWRIGHT_DEBUG`? We might want to guard other debugging features behind this env.

File: src/chromium/crDevTools.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        result = this._prefs;
+      } else if (parsed.method === 'setPreference') {
+        this._prefs[parsed.params[0]] = parsed.params[1];
+        this._save();
+      } else if (parsed.method === 'removePreference') {
+        delete this._prefs[parsed.params[0]];
+        this._save();
+      } else if (parsed.method === 'clearPreferences') {
+        this._prefs = {};
+        this._save();
```
Comment (aslushnikov) [inline src/chromium/crDevTools.ts:57]: can we serialize these saves? I'm afraid they'll race against each other.

**Review by aslushnikov — APPROVED** [review]

File: src/chromium/crDevTools.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    await Promise.all([
+      session.send('Runtime.enable'),
+      session.send('Runtime.addBinding', { name: kBindingName }),
+      session.send('Page.enable'),
+      session.send('Page.addScriptToEvaluateOnNewDocument', { source: `
+        (() => {
+          const init = () => {
+            // Lazy init happens when InspectorFrontendHost is initialized.
+            // At this point DevToolsHost is ready to be used.
+            const host = window.DevToolsHost;
```
[the remark this answers is not in the document]
Comment (dgozman) [reply src/chromium/crDevTools.ts:78]: Done.

File: src/chromium/crDevTools.ts (hunk 2 above)
Comment (dgozman) [reply src/chromium/crDevTools.ts:62]: Done.

**Pull request merged**